ALTER TABLE CLUSTER commands should not be supported for AO tables.

jimmyyih opened the pull request:
Append-optimized tables do not support CLUSTER command. For some
unknown reason, the ALTER TABLE..CLUSTER and ALTER TABLE..SET WITHOUT
CLUSTER commands were left alone. This commit will properly disable
both commands for append-optimized (row and column orientated) tables
and report an error message when used.

hlinnaka commented:
There's not much need to forbid ALTER TABLE SET WITHOUT CLUSTER. It won't do anything, because you cannot set a clustering order in the first place, but I don't see much harm in allowing it. Kind of like "DROP IF EXISTS"

jimmyyih commented:
Agreed that it doesn't do much, but I mainly added the change to ALTER TABLE SET WITHOUT CLUSTER for strict consistency and as a reminder to users that all CLUSTER operations are not supported for AO tables.  Although I kinda agree now that there's not much harm in allowing the no op.

@Eulerizeit, @asimrp: Should we allow ALTER TABLE SET WITHOUT CLUSTER on AO tables? It may be good to keep..... for like maybe when a user upgrades and for some reason has cluster on an AO index and wishes to remove the cluster on the index to conform.

Eulerizeit commented:
This seems like a good question for the gpdb-users email. I don't care, we should choose whatever behavior impacts users the least badly or the most positively.

jimmyyih commented:
Created gpdb-users thread (https://groups.google.com/a/greenplum.org/forum/#!topic/gpdb-users/y72Ze8MEZR4) but it didn't get any traction.  I have removed the restriction on ALTER TABLE SET WITHOUT CLUSTER and added doc updates.

@mkiyama, can you please review the doc update?

mkiyama commented on gpdb-doc/dita/ref_guide/sql_commands/ALTER_TABLE.xml:
```diff
@@ -180,7 +180,9 @@ ALTER TABLE <varname>name</varname>
                                                   <codeph><xref href="./CREATE_TABLE_AS.xml#topic1"
                                                   type="topic" format="dita"/></codeph> and order it
-                                        by the index column(s).</li>
+                                        by the index column(s).
```
I would say 
~~The~~ CLUSTER ON ~~action~~ is not supported with append-optimized tables.

jimmyyih replied:
Got it, thanks!
